src/Sema.zig

```diff
@@ -24295,6 +24302,9 @@ fn zirSplat(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError!Air.I
 
     if (!dest_ty.isVector(mod)) return sema.fail(block, src, "expected vector type, found '{}'", .{dest_ty.fmt(mod)});
 
+    if (!dest_ty.hasRuntimeBits(mod))
+        return Air.internedToRef((try sema.splat(dest_ty, null)).toIntern());
```

Comment:
You wouldn't need to complicate `splat` if you directly returned the empty aggregate here.

Reply:
done!